feat: RFD 8 Public Roadmap (#1433)
cloudfoundry-incubator/kubecf: doc/rfd/rfd/0008/README.md

---
authors: Jaime Gomes <jaime.gomes@suse.com>
state: discussion
discussion: https://github.com/cloudfoundry-incubator/kubecf/issues?q=%22RFD+8%22+
---

# RFD 0008 Public Roadmap

## Motivation

The *Roadmap* project can be used used to track the features that the team decided to engage with,

> which phase they are and when they may come available.

Reviewer: ```suggestion
which phase they are in and when they may come available.
```

The goal is to provide to the community a glimpse to our main features, the problem that they're
solving and how the team is planning to solve.

## Guideline

## Submit the features

New features aren't added directly to the *Roadmap* project but instead added as an issue into the
*KubeCF* default project with the label *Type: Enhancement*.

Note: when opening a new issue do not forget please use the right template.

## Phases

Every project feature has it's own lifecycle.

* Beta: public available but for testing and gather feedback.
* GA: public available.

Some features may be allocated to a desired timeframe but still in exploratory phases, which can

> lead to potential delays, since their main purpose is to gather community feedback.

Author: Even for me ;)

I am rephrasing it....

* in design: we have decided to build but still figuring out how.
* exploring: we are considering building, but we need more feedback to make a final decision.

> Any feature can be abandoned by team decision, independently of it state and the reason well

Reviewer 2: ```suggestion
Any feature can be abandoned by team decision independently of its state, provided there's a reason well
```

well-documented reason.

## Stages

The roadmap is arranged in quarters columns to give an idea when is our intention to release that

> particular feature.

Reviewer: Rephrased:
```
The roadmap is arranged in columns, each representing a quarter year.
The contents indicate our intent on when to release any particular feature.
Note that an intent is not a promise.
```

> intent to release a feature.

Reviewer: ```
- ... it indicates when are we intent to release a feature.
+ .. it indicates when we intend to release a feature.
```
`intent` is the thing/noun, `(to) intend` the action/verb.

We also have two labels: *in design* or *exploring* which in conjunction with the timeframe column
can indicate how far are we with that particular feature and it may give a better sense if it can be
released within the initial timeframe.

## Disclaimer

This roadmap does not represent any commitment, promise or obligation to deliver any feature or any

> feature on a particular date. The main goal is to give the community the team high level development

Reviewer: ```suggestion
feature on a particular date. The main goal is to show the community the team's high level development
```

> plan, so please do not rely on this roadmap to make any internal decision.

Reviewer: ```suggestion
plan. Please __do not__ rely on this roadmap to make any internal decision.
```
